src/cmd/build.js

```diff
@@ -60,8 +61,21 @@ function defaultPackageCreator(
           filter: (...args) => fileFilter.wantFile(...args),
         })
         .then((buffer) => {
+          let messageData: any;
+          if (manifestData.default_locale) {
+            messageData = readFileSync(
+               path.join(sourceDir, '_locales', manifestData.default_locale,
+                'messages.json'));
+          }
+          let extensionName: string;
+          if (messageData) {
+            extensionName = JSON.parse(messageData).extensionName.description;
```

Comment:
I think we could maybe move it there in the future. It will be easier to review a patch with a standalone function first though. The problem with adding it to `getValidatedManifest()` is not every part of jpm needs this localization code (which does some I/O so it's not cheap). We'll probably need to return an object and give it a method like `manifest.getExtensionName()`.